etc/testing/circle/workloads/go-test-results/egress/main.go

```diff
@@ -0,0 +1,194 @@
+	inputFolder := os.Args[1]
+
+	db, err := sqlx.Open("pgx", fmt.Sprintf("user=%s password=%s host=%s port=5432 database=ci_metrics", postgresqlUser, postgresqlPassword, postgresqlHost))
+	if err != nil {
+		logger.Fatalf("Failed opening db connection %v", err)
+	}
+	// filpathWalkDir does not evaluate the PFS symlink
+	sym, err := filepath.EvalSymlinks(inputFolder)
+	if err != nil {
+		logger.Fatal(err)
+	}
+	jobInfoPaths := make(map[string]gotestresults.JobInfo)
+	testResultPaths := []string{}
+	err = filepath.WalkDir(sym, func(path string, d fs.DirEntry, err error) error {
+		if !d.IsDir() {
+			if d.Name() == jobInfoFileName {
+				jobInfo, err := readJobInfo(path)
+				if err != nil {
+					return err
+				}
+				return insertJobInfo(jobInfo, jobInfoPaths, path, d, db)
+			} else {
+				// we need the job info inserted first due to foreign key constraints,
+				// so save results to loop over later ensuring ordered inserts
+				testResultPaths = append(testResultPaths, path)
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		logger.Fatal("Walking file tree for job info ", err)
+	}
+	if len(jobInfoPaths) == 0 {
+		logger.Println("No new job info found, exiting without inserting test results.")
+		return
+	}
+	for _, path := range testResultPaths {
+		if err = insertTestResultFile(path, jobInfoPaths, db); err != nil {
+			logger.Fatal("Inserting test results into DB ", err)
+		}
+	}
+}
+
+func readJobInfo(path string) (*gotestresults.JobInfo, error) {
+	jobInfoFile, err := os.Open(path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling job info json ")
+	}
+	defer jobInfoFile.Close()
+	jobInfoJson, err := io.ReadAll(jobInfoFile)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling job info json ")
+	}
+	var jobInfo gotestresults.JobInfo
+	if err = json.Unmarshal(jobInfoJson, &jobInfo); err != nil {
+		return nil, errors.Wrapf(err, "unmarshalling job info json ")
+	}
+	return &jobInfo, nil
+}
+
+func insertJobInfo(jobInfo *gotestresults.JobInfo, jobInfoPaths map[string]gotestresults.JobInfo, path string, d fs.DirEntry, db *sqlx.DB) error {
+	_, err := db.Exec(`INSERT INTO public.ci_jobs (
+							workflow_id, 
+							job_id, 
+							job_executor,
+							job_name, 
+							job_timestamp, 
+							job_num_executors,
+							commit_sha, 
+							branch, 
+							tag,
+							pull_requests
+						) VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)`,
+		jobInfo.WorkflowId,
+		jobInfo.JobId,
+		jobInfo.JobExecutor,
+		jobInfo.JobName,
+		jobInfo.JobTimestamp,
+		jobInfo.JobNumExecutors,
+		jobInfo.Commit,
+		jobInfo.Branch,
+		jobInfo.Tag,
+		jobInfo.PullRequests,
+	)
+	if err != nil {
+		var pgErr *pgconn.PgError
+		if errors.As(err, &pgErr) && pgErr.Code == "23505" {
+			// Unique key constraint violation. In other words, the record already exists.
+			// Ignore insert to since we don't want to change the timestamp, but log it. This is expected with multiple executors.
```

Comment:
This comment is outdated but close - should be updated now that the executors are a folder.

Reply:
In your SQL you can write "ON CONFLICT DO ...".  The ... can be NOTHING if you don't care, or an update statement if you want to do an update instead of an insert.  It's not a big deal to do it in Go, but Postgres can save you some trouble if you want.